hail/python/hailtop/aiotools/fs/copier.py

```diff
@@ -389,9 +393,20 @@ async def copy_source(srcentry):
 
             await self._copy_file_multi_part(sema, source_report, srcfile, await srcentry.status(), url_join(full_dest, relsrcfile), return_exceptions)
 
-        await bounded_gather2(sema, *[
-            functools.partial(copy_source, srcentry)
-            async for srcentry in srcentries], cancel_on_error=True)
+        async def create_copies() -> List[Callable[[], Awaitable[None]]]:
+            nonlocal srcentries
+            if srcentries is None:
+                srcentries = await files_iterator()
+            try:
+                return [
+                    functools.partial(copy_source, srcentry)
+                    async for srcentry in srcentries]
+            finally:
+                srcentries = None
+
+        copies = await retry_transient_errors(create_copies)
+
+        await bounded_gather2(sema, *copies, cancel_on_error=True)
```

Comment:
Just to be clear, the purpose of this change is to retry transient errors when listing the files to figure out what to copy?

Reply:
sorry. just saw the title of the PR.

Reply:
Yep!